pkg/resources/fluentd/fluentd.go

```diff
@@ -101,6 +124,11 @@ func New(client client.Client, log logr.Logger,
 
 // Reconcile reconciles the fluentd resource
 func (r *Reconciler) Reconcile(ctx context.Context) (*reconcile.Result, error) {
+	// No fluentdSpec yet, explicitly ask for a requeue to short circuit the controller loop after the status update
```

Comment:
this is unnecessary as fluentdSpec will always exist at this point, also we leverage the watch instead of having to requeue